src/client/kv/dc_kv.c

```diff
@@ -67,22 +68,23 @@ dc_kv_put(tse_task_t *task)
 {
 	daos_kv_put_t		*args = daos_task_get_args(task);
 	daos_obj_update_t	*update_args;
-	tse_task_t		*update_task;
-	struct io_params	*params;
+	tse_task_t		*update_task = NULL;
+	struct io_params	*params = NULL;
 	int			rc;
 
+	if (args->key == NULL || args->buf_size == 0 || args->buf == NULL)
+		D_GOTO(err_task, rc = -DER_INVAL);
+
 	D_ALLOC_PTR(params);
-	if (params == NULL) {
-		D_ERROR("Failed memory allocation\n");
-		return -DER_NOMEM;
-	}
+	if (params == NULL)
+		D_GOTO(err_task, rc = -DER_NOMEM);
 
 	/** init dkey */
 	d_iov_set(&params->dkey, (void *)args->key, strlen(args->key));
 
-	/** init iod. For now set akey = dkey */
-	d_iov_set(&params->iod.iod_name, (void *)args->key,
-		     strlen(args->key));
+	/** init iod. */
+	params->akey_str = '0';
```

Comment:
this should really be changed to NULL when it's supported for this truly to be a flat KV.
i will change the name of the param to avoid misguidance as Li Wei also suggested.